planner/core/task.go

```diff
@@ -1114,6 +1115,13 @@ func CheckAggCanPushCop(sctx sessionctx.Context, aggFuncs []*aggregation.AggFunc
 			return false
 		}
 		if !aggregation.CheckAggPushDown(aggFunc, storeType) {
+			if sc.InExplainStmt {
+				storageName := storeType.Name()
+				if storeType == kv.UnSpecified {
+					storageName = "storage layer"
```

Comment:
when will we encounter this branch?

Reply:
In `predicate_push_down` rule, it uses `Unspecified` to make sure it can pushdown all the possible expressions, and when generating the cop task, the `DataSource` will use the specific store type(`TiFlash/TiKV/TiDB`) to make sure only pushdown supported expressions to each store.